### DISCUSSION TITLE
How to enable external web access when installed on a Synology home server

### DISCUSSION THREAD

**Discussion by cosmokramer99** [teslamate-org/teslamate#2275] [Q&A]
I've gotten Teslamate up and running on my home NAS no problem.  However, it is only available on my home network.  In theory I could get a static domain name that could point to the home server (installing DNS Server on NAS, etc) but still learning and not sure if this is really possible?

**Comment by bahree** [comment]
Synology has a DDNS option out of the box. If you goto Control Panel --> External Access there is a option on DDNS where you can add a few - including free ones. See the screenshot below.
![image](https://user-images.githubusercontent.com/3529468/148149566-7e47dbb3-23c6-478e-ab86-71fa052dc6dc.png)

And for connection outside of your home network, you will need to setup port fwd'ing from your router to the NAS and also ensure its enabled in Router Configuration as part of External Access.
![image](https://user-images.githubusercontent.com/3529468/148149823-cc8e3772-b0d1-4ef7-a9b3-ca33a1e75395.png)

However, opening the NAS has other issues and potentially opens the surface area for attack; you need to ensure you lock it down properly including account protection and other things. Another approach to consider would be to run this in the cloud and expose it there. One example I show how-to run-on Azure here: https://www.desigeek.com/blog/amit/2021/12/30/how-to-run-teslamate-on-azure/

**Comment by cosmokramer99** [comment]
Thanks for the reply.  I think I’ve resigned to run in cloud for external access due to potential security issues you outlined.  Appreciated.

**Reply by sheenhx** [reply]
or you just simply can do a VPN to your NAS and check it , much safer
